src/Whitelist/VulnerabilityExcluder.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { existsSync, readFileSync } from "fs";
+import path from 'path';
+
+import { OssIndexServerResult } from "../Types/OssIndexServerResult";
+
+const whitelistFilePathPwd = path.join(process.cwd(), 'auditjs.json');
+
+export abstract class VulnerabilityExcluder {
+  public static filterVulnerabilities(results: Array<OssIndexServerResult>, whitelistFilePath: string = whitelistFilePathPwd): Array<OssIndexServerResult> {
+    if (existsSync(whitelistFilePath)) {
+      const json = readFileSync(whitelistFilePath);
+
+      const whitelist = JSON.parse(json.toString());
```

Comment:
related to the point I made in [this comment]():

`JSON.parse` would throw an exception if trying to parse an invalid json so you'd need to wrap this in a try/catch.

```
let whitelist;
try {
  whitelist = JSON.parse(json.toString());
}
catch (exception) {
  console.error('Could not parse json file'); // or whatever Logger you're using
}
```

And then there's the question of what to do if there isn't a valid whitelist json? return `results`?

Reply:
I decided to `throw new Error()` in this case, returning results seems like it could get ignored by someone, potentially putting them in a dangerous situation where they are like:

![this_is_fine](https://user-images.githubusercontent.com/5544326/71632909-80ec3b00-2bbd-11ea-9631-a981251dc536.jpg)

I'd rather they get the error and have to fix things.

Reply:
Everything put together in: fd46cb8 and 67220b7